msticpy/data/uploaders/splunk_uploader.py

```diff
@@ -0,0 +1,252 @@
+        progress = tqdm(total=len(data.index), desc="Rows", position=0)
+        for row in data.iterrows():
+            data = row[1].to_csv()
+            try:
+                data.encode(encoding="latin-1")
+            except UnicodeEncodeError:
+                data = data.encode(encoding="utf-8")
+            index.submit(data, sourcetype=table_name, host=host)
+            progress.update(1)
+        progress.close()
+        if self._debug is True:
+            print("Upload complete")
+
+    # pylint: disable=arguments-differ
+    def upload_df(  # type: ignore
+        self, data: pd.DataFrame, table_name: str, index_name: str, **kwargs,
+    ):
+        """
+        Upload a Pandas DataFrame to Splunk.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            Data to upload.
+        table_name : str
+            The souretype in Splunk data will be uploaded to.
+        index_name : str
+            Name of the Splunk Index to add data to, if it doesn't exist it will be created.
+        host : str, optional
+            Host name to upload data with, default will be 'Upload'
+
+        """
+        host = kwargs.get("host", None)
+        create_idx = kwargs.get("create_index", "False")
+        if not isinstance(data, pd.DataFrame):
+            raise MsticpyUserError(
+                "Data must be in Pandas DataFrame format.",
+                title="incorrect data format",
+            )
+        self._post_data(
+            data=data,
+            table_name=table_name,
+            index_name=index_name,
+            create_index=create_idx,
+            host=host,
+        )
+
+    def upload_file(  # type: ignore
+        self,
+        file_path: str,
+        index_name: str,
+        table_name: str = None,
+        delim: str = ",",
+        **kwargs,
+    ):
+        """
+        Upload a seperated value file to Splunk.
+
+        Parameters
+        ----------
+        file_path : str
+            Path to the file to upload.
+        index_name : str
+            Name of the Splunk Index to add data to, if it doesn't exist it will be created.
+        table_name : str, optional
+            The souretype in Splunk data will be uploaded to, if not set the file name will be used.
+        delim : str, optional
+            Seperator value in file, by default ","
+        host : str, optional
+            Host name to upload data with, default will be 'Upload'
+
+        """
+        host = kwargs.get("host", None)
+        create_idx = kwargs.get("create_index", "False")
+        path = Path(file_path)
+        try:
+            data = pd.read_csv(path, delimiter=delim)
+        except (ParserError, UnicodeDecodeError):
+            raise MsticpyUserError(
+                "The file specified is not a seperated value file.",
+                "Incorrect file type.",
+            )
+
+        if not table_name:
+            table_name = str(path).split("\\")[-1].split(".")[0]
```

Comment:
Should use Path from pathlib. This handles different OS separators. You can also get the file name minus extension (stem? maybe)